canonical-web-and-design/maas-ui
# refactor(ui): replace old Pod components with new KVM components

## Done

- Removed the old `Pod` components that did not use the `resources` parameters
- Renamed `LxdResources` to `KVMResources` and updated it so it works with both virsh and LXD pods (turns out the differences were smaller than I thought).
- Renamed `PodStorage` to `KVMStorageCards` and moved under `KVMResources` since it only shows up in that tab

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and click on a virsh KVM.
- Check that there is no "Overall" card like with LXD.
- Check that the resources card renders properly and does not show a project name. If the pod has NUMA nodes you should be able to toggle the NUMA view.

## Fixes

Fixes #2476
